### PR TITLE
:sparkles: Add support for simple wilcard Anchor.toml's workspace.members

### PR DESCRIPTION
**Pull request by GabrielePicco**
# Add support for simple wilcard Anchor.toml's workspace.members

See #2784 and #2658 for context

### CODE REVIEW

File: cli/src/config.rs [hunk 1]
```diff
@@ -304,15 +304,25 @@ impl WithPath<Config> {
...
+                if m.ends_with("/*") {
+                    let dir = path.parent().unwrap();
+                    match fs::read_dir(dir) {
+                        Ok(entries) => entries
+                            .filter_map(|entry| entry.ok())
+                            .map(|entry| entry.path().canonicalize().unwrap_or_else(|_| {
+                                panic!("Error canonicalizing path {:?}", entry.path());
+                            }))
+                            .collect::<Vec<_>>()
+                            .into_iter(),
```
Comment (acheroncrypto) [inline cli/src/config.rs:318]: Turbofish is not required here and `Vec` implements `IntoIterator` so there is no need to call it manually.

```suggestion
                            .collect(),
```

**Review by acheroncrypto — COMMENTED** [review]
Thanks, this should work but can we get rid of explicit `panic!`s? I'm aware the previous code also had it but let's not add more of it since this method already returns a `Result` so that the error can be handled gracefully.

It would also be great to have this for `workspace.exclude`:

https://github.com/coral-xyz/anchor/blob/ef3b1493484e9d193f3cb270753484c30f75bcd8/cli/src/config.rs#L318-L332

**Comment by GabrielePicco** [comment]
@acheroncrypto I removed the `panic!`s and refactored the code so that it's also applied to `workspace.exclude`

**Review by acheroncrypto — COMMENTED** [review]
Looks great!

Before we merge, could you note this change in the [CHANGELOG](https://github.com/coral-xyz/anchor/blob/ef3b1493484e9d193f3cb270753484c30f75bcd8/CHANGELOG.md#unreleased)?

**Comment by GabrielePicco** [comment]
Sure. Noted in the changelog.

**Review by acheroncrypto — APPROVED** [review]
Thank you!

**Pull request merged**